### ISSUE TITLE
[docdb] Tablet splitting: Enhance TS side throttling to not have too many splits in progress

### ISSUE THREAD

**Issue by bmatican** [yugabyte/yugabyte-db#7917]
The TS reports tablets to split to the master, via heartbeats. Currently, this happens 1 tablet at a time, until said tablet is split.

However, currently, we consider a tablet split, as soon as the children are online. But afterwards, we schedule a compaction on both children, which could be IO and CPU intensive and take quite a bit of time.

In the meantime, the TS might report the next tablet to the master and thus trigger a new split, and this can cascade into many tablet splits needing to run background compactions.

One easy thing we could do is to only consider the split done, for TS>master reporting, once the background compaction is done. This would minimize the impact on each TS.

cc @ttyusupov @mikhpolitov @robertsami

**Comment by ttyusupov** [comment]
See https://github.com/yugabyte/yugabyte-db/issues/6667

**Comment by bmatican** [comment]
no longer relevant now that we have master side control
